packages/firestore/lib/modular/index.js

```diff
@@ -0,0 +1,222 @@
+ */
+export function collection(parent, path, ...pathSegments) {
+  if (pathSegments && pathSegments.length) {
+    path = path + '/' + pathSegments.map(e => e.replace(/^\/|\/$/g, '')).join('/');
+  }
+
+  return parent.collection(path);
+}
+
+/**
+ * @param {Firestore} firestore
+ * @param {string} collectionId
+ * @returns {Query<DocumentData>}
+ */
+export function collectionGroup(firestore, collectionId) {
+  return firestore.collectionGroup(collectionId);
+}
+
+/**
+ * @param {DocumentReference} reference
+ * @param {import('.').PartialWithFieldValue} data
+ * @param {SetOptions?} options
+ * @returns {Promise<void>}
+ */
+export function setDoc(reference, data, options) {
+  return reference.set(data, options);
+}
+
+/**
+ * @param {DocumentReference} reference
+ * @param {string | FieldPath | import('.').UpdateData} fieldOrUpdateData
+ * @param {unknown?} value
+ * @param {unknown} moreFieldsAndValues
+ * @returns {Promise<void>}
+ */
+export function updateDoc(reference, fieldOrUpdateData, value, ...moreFieldsAndValues) {
+  if (!fieldOrUpdateData) {
+    // @ts-ignore
+    return reference.update();
+  }
+
+  if (!value) {
+    return reference.update(fieldOrUpdateData);
+  }
+
+  if (!moreFieldsAndValues || !Array.isArray(moreFieldsAndValues)) {
+    return reference.update(fieldOrUpdateData, value);
+  }
+
+  return reference.update(fieldOrUpdateData, value, ...moreFieldsAndValues);
+}
+
+/**
+ * @param {CollectionReference} reference
+ * @param {WithFieldValue} data
+ * @returns {Promise<DocumentReference>}
+ */
+export function addDoc(reference, data) {
+  return reference.add(data);
+}
+
+/**
+ * @param {Firestore} firestore
+ * @returns {Promise<void>}
+ */
+export function enableNetwork(firestore) {
+  return firestore.enableNetwork();
+}
+
+/**
+ * @param {Firestore} firestore
+ * @returns {Promise<void>}
+ */
+export function disableNetwork(firestore) {
+  return firestore.disableNetwork();
+}
+
+/**
+ * @param {Firestore} firestore
+ * @returns {Promise<void>}
+ */
+export function clearPersistence(firestore) {
+  return firestore.clearPersistence();
+}
+
+/**
+ * @param {Firestore} firestore
+ * @returns {Promise<void>}
+ */
+export function terminate(firestore) {
+  return firestore.terminate();
+}
+
+/**
+ * @param {Firestore} firestore
+ * @returns {Promise<void>}
+ */
+export function waitForPendingWrites(firestore) {
+  return firestore.waitForPendingWrites();
+}
+
+/**
+ * @param {FirebaseApp} app
+ * @param {FirestoreSettings} settings
+ * @param {string?} databaseId
+ * @returns {Promise<Firestore>}
+ */
+// FIXME: databaseId is not supported by compat API?
+// FIXME: as per web SDK, initializeFirestore would override existing firestore instance.
+//       Calling getFirestore() after initializeFirestore() would return this new instance.
+//       Need to figure out how to handle this using the compat API.
+export async function initializeFirestore(app, settings, databaseId) {
+  databaseId; // make eslint happy
```

Comment:
maybe make a note that this needs to be implemented once 2nd database for Firestore is supported.